src/meta/src/stream/scale.rs

```diff
@@ -539,6 +539,17 @@ where
                     );
                 }
             }
+
+            if current_parallel_units
+                .iter()
+                .all(|x| removed_parallel_units.contains(x))
```

Comment:
It should be valid if `current = [0,1,2], removed = [0,1,2], added = [3]`.

After L566, we're sure that the inputs are valid. So I guess we only need to check that `current.len() + added.len() > removed.len()` for `Hash` fragments.